pep-9999.rst

```diff
@@ -81,9 +81,201 @@ includes stubs for Python's standard library and several third-party
 packages. These are usually distributed with type checkers and do not
 require separate installation.
 
+Style Guide
+===========
+
+Stub files should generally follow the Style Guide for Python Code (PEP 8)
+[#pep8]_. There are a few exceptions, outlined below, that take the
+different structure of stub files into account and are aimed to create
+more concise files.
+
+Maximum Line Length
+-------------------
+
+Type stubs should be limited to 130 characters per line.
+
+Blank Lines
+-----------
+
+Do not use empty lines between functions, methods, and fields, except to
+group them with one empty line. Use one empty line around classes, but do not
+use empty lines between body-less classes, except for grouping.
+
+Yes::
+
+    def time_func() -> None: ...
+    def date_func() -> None: ...
+
+    def ip_func() -> None: ...
+
+    class Foo:
+        x: int
+        y: int
+        def __init__(self) -> None: ...
+
+    class MyError(Exception): ...
+    class AnotherError(Exception): ...
+
+No::
+
+    def time_func() -> None: ...
+
+    def date_func() -> None: ...  # do no leave unnecessary empty lines
+
+    def ip_func() -> None: ...
+
+
+    class Foo:  # leave only one empty line above
+        x: int
+    class MyError(Exception): ...  # leave an empty line after the class
+
+Module Level Dunder Names
+-------------------------
+
+Do not include ``__all__`` in type stubs.
+
+Module Level Attributes
+-----------------------
+
+Do not use an assignment for module-level attributes.
+
+Yes::
+
+    CONST: str
+    x: int
+
+No::
+
+    CONST = ""
+    x: int = 0
+    y: float = ...
+    z = 0  # type: int
+
+Classes
+-------
+
+Do not include docstrings in class bodies.
+
+Classes without bodies should use the ellipsis operator ``...`` in place
+of the body on the same line as the class definition.
+
+Yes::
+
+    class MyError(Exception): ...
+
+No::
+
+    class MyError(Exception):
+        ...
+    class AnotherError(Exception): pass
+
+Functions and Methods
+---------------------
+
+The bodies of functions and methods should consist of only the ellipsis
+operator ``...`` on the same line as the closing parenthesis and colon.
+Do not include docstrings.
+
+Yes::
+
+    def to_int1(x: str) -> int: ...
+    def to_int2(
+        x: str,
+    ) -> int: ...
+
+No::
+
+    def to_int1(x: str) -> int:
+        return int(x)
+    def to_int2(x: str) -> int:
+        ...
+    def to_int3(x: str) -> int: pass
+
+Use the ellipsis operator ``...`` in place of actual default argument
+values. Use an explicit ``Optional`` annotation instead of
+a ``None`` default.
+
+Yes::
+
+    def foo(x: int = ...) -> None: ...
+    def bar(y: Optional[str] = ...) -> None: ...
+
+No::
+
+    def foo(x: int = 0) -> None: ...
+    def bar(y: str = None) -> None: ...
+    def baz(z: Optional[str] = None) -> None: ...
+
+Private Classes
+---------------
+
+If a function or method returns an instance of a private class, the
```

Comment:
Why public? Shouldn't this be a private class?

Reply:
True, but I don't see a better solution for now. Maybe we could add an example of this to the PEP, though?

Reply:
I think using a private class is fine actually, and that's what we've been doing in typeshed so far. It should be rare that a user needs to use the class in a type annotation.

Reply:
One very common case and good example are the pattern and match classes of the `re` module, although they are of course special cased by `typing`.

Reply:
One case where a private class is needed as a type annotation is a wrapper around a library method that returns an instance of the private class. I'm not sure how rare that is, and honestly a class that is returned as part of an API should not be private.